bootstrap/config/config.go

```diff
@@ -197,7 +198,7 @@ func (cp *Processor) Process(
 		// NOTE: Some security services don't use any common configuration and don't use the configuration provider.
 		commonConfigLocation := environment.GetCommonConfigFileName(cp.lc, cp.flags.CommonConfig())
 		if commonConfigLocation != "" {
-			err := cp.loadCommonConfigFromFile(commonConfigLocation, serviceConfig, serviceType)
+			err := cp.loadCommonConfigFromFile(commonConfigLocation, serviceConfig, serviceType, secretProvider)
```

Comment:
We don't need to be passing secretProvider in all this function calls.

loadConfigYamlFromFile can get it from the DIC.